api/lib/infrastructure/repositories/assessment-repository.js

```diff
@@ -87,10 +87,19 @@ module.exports = {
       .then(_toDomain);
   },
 
-  getByAssessmentIdAndUserId(assessmentId, userId) {
+  getByAssessmentIdAndUserId({ assessmentId, userId }) {
     return BookshelfAssessment
       .query({ where: { id: assessmentId }, andWhere: { userId } })
-      .fetch({ require: true })
+      .fetch({
+        withRelated: [
+          {
+            answers: function(query) {
+              query.orderBy('createdAt', 'ASC');
+            },
+          }, 'assessmentResults', 'campaignParticipation', 'campaignParticipation.campaign',
+        ],
+        require: true
+      })
```

Comment:
Je vois pas trop le rapport entre le nom de la PR et ce changement (qui est conséquent) ?

Reply:
Avant on avait un `get-assessment` coté usecase qui faisait juste ça à savoir récup l'assessment, et maintenant si je comprend bien 
- la variable qui récup l'assessment se nomme `assessmentWithResults`
- sauf que y'a plus que les `assessment-results` mais y'a d'autres trucs 
- et le usecase s'appelle tjrs `get-assessment` ?

Reply:
Coté usecase, il appelait `get` du repository, qui renvoie assessmentResults, etc.
Le pré-handler appelait `getByAssessmentIdAndUserId` qui lui ne renvoyait pas toutes les informations.
Le usecase serait plutot maintenant : improve-assessment-with-level-and-score si on veut être clair

Reply:
@Radioreve je ne vois pas de raison que le _usecase_ change de nom sous prétexte que son implémentation évolue (en l’occurence parce que des données sont récupérées par ailleurs). Du point de vue de l’utilisateur il réalise toujours la même fonction.

Sur cette PR en général je trouve que les questions qui se posent autour de la récupération des « bonnes » données dans le _prehandler_ tendent à indiquer que ce n’est peut-être pas si rentable que ça d’avoir le _prehandler_ qui transmet les données au _usecase_ via le _controller_. Certes ça permet d’économiser une requête dans certains cas mais au prix d’une confusion dans les responsabilités et les frontières architecturales.

Ma suggestion serait de ne pas transférer de données du _prehandler_ au _controller_ (ou alors juste `userId` et `assessmentId`). Ou bien, de ne pas utiliser de _prehandler_ pour sécuriser ces routes, parce que ça impose le découpage de l’infrastructure Web au code métier sous-jacent.